src/figuresClasses.ts

```diff
@@ -1,11 +1,82 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
```

Comment:
[CODE KNOWLEDGE] - There's no need to manually assign the constructor parameters to class properties since TypeScript does this automatically when you use the 'public' keyword. You can remove the assignments to 'this.color', 'this.a', 'this.b', and 'this.c' in the constructor.